src/main/kotlin/com/celuveat/auth/adaptor/out/TokenAdaptor.kt

```diff
@@ -0,0 +1,54 @@
+package com.celuveat.auth.adaptor.out
+
+import com.celuveat.auth.application.port.out.CreateTokenPort
+import com.celuveat.auth.application.port.out.ExtractClaimPort
+import com.celuveat.auth.domain.Token
+import com.celuveat.common.annotation.Adapter
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.io.Decoders
+import io.jsonwebtoken.security.Keys
+import java.util.*
+import javax.crypto.SecretKey
+
+@Adapter
+class TokenAdaptor(
+    tokenProperty: TokenProperty,
+) : CreateTokenPort, ExtractClaimPort {
+
+    private val secretKey: SecretKey = Keys.hmacShaKeyFor(Decoders.BASE64URL.decode(tokenProperty.secretKey))
+    private val accessTokenExpirationMillis: Long = tokenProperty.accessTokenExpirationMillis
+
+    override fun create(key: String, claim: String): Token {
+        return create(mapOf(key to claim))
+    }
+
+    override fun create(claims: Map<String, String>): Token {
+        return Token(
+            Jwts.builder()
+                .claims(claims)
+                .issuedAt(Date())
+                .expiration(Date(System.currentTimeMillis() + accessTokenExpirationMillis))
+                .signWith(secretKey, Jwts.SIG.HS512)
+                .compact()
+        )
+    }
+
+    override fun extract(token: String, key: String): String {
+        return extract(token)[key]
+            ?: throw NoSuchElementException("Claim '$key' not found in token")  // TODO 예외처리
+    }
+
+    override fun extract(token: String): Map<String, String> {
+        try {
+            return Jwts.parser()
+                .verifyWith(secretKey)
+                .build()
```

Comment:
별 차이는 없는데 아래처럼 외부에서 반환해도 좋을 것 같아
```suggestion
        return try {
            Jwts.parser()
                .verifyWith(secretKey)
                .build()
```

Reply:
오.. 되는지 처음 알았어..
자바도 되나 저거